### ISSUE TITLE
has_one :through eager loading using includes creates invalid SQL

### ISSUE THREAD

**Issue by ghost** [rails/rails#14046]
In the case, where `has_one :through` refers to a model that is a child class, `includes` causes AR to create invalid SQL.

My gist is an example of this behaviour: https://gist.github.com/mp-dhorsak/8977542

Error: 

```
ActiveRecord::StatementInvalid: SQLite3::SQLException: no such column: geographies.type: SELECT "addresses".* FROM "addresses"  WHERE "geographies"."type" IN ('Country') AND "addresses"."user_id" IN (1)
```

The problem occurs in both Rails 4.0.2 and 4.1.0.beta1

**Comment by magikid** [comment]
It looks @ghost deleted their account and the associated gist.  Does anyone still have reproduction steps?

**Comment by arturmalecki** [comment]
Sorry, this was 2 years ago. Don't remember how to reproduce it.

**Comment by magikid** [comment]
Ok, it looks like #

**Comment by alexcameron89** [comment]
I was able to reproduce the same problem here through `git bisect` using the script from #12114. This issue was fixed in Rails 5 when Arel 7.0.0.alpha was introduced, 2223f49.
